ignite/metrics/gan/fid.py

```diff
@@ -0,0 +1,177 @@
+
+    diff = mu1 - mu2
+    import scipy
+
+    # Product might be almost singular
+    covmean, _ = scipy.linalg.sqrtm(sigma1.mm(sigma2), disp=False)
+    if not np.isfinite(covmean).all():
+        offset = np.eye(sigma1.shape[0]) * eps
+        covmean = scipy.linalg.sqrtm((sigma1 + offset).dot(sigma2 + offset))
+
+    # Numerical error might give slight imaginary component
+    if np.iscomplexobj(covmean):
+        if not np.allclose(np.diagonal(covmean).imag, 0, atol=1e-3):
+            m = np.max(np.abs(covmean.imag))
+            raise ValueError("Imaginary component {}".format(m))
+        covmean = covmean.real
+
+    tr_covmean = np.trace(covmean)
+
+    return diff.dot(diff) + np.trace(sigma1) + np.trace(sigma2) - 2 * tr_covmean
+
+
+class InceptionExtractor:
+    def __init__(self) -> None:
+        from torchvision import models
+
+        self.model = models.inception_v3(init_weights=True)
+        self.model.fc = torch.nn.Sequential()
+        self.model.eval()
+
+    def __call__(self, data: torch.Tensor) -> torch.Tensor:
+        if data.shape[1:] != (3, 299, 299):
+            raise ValueError(f"Images should be of size 3x299x299 (got {data.shape})")
+        return self.model(data).detach()
+
+
+class FID(Metric):
+    r"""Calculates Frechet Inception Distance.
+
+    .. math::
+       \text{FID} = \mod{\text{mu1} - \text{mu2}} + \text{Trace(sigma1 + sigma2 - 2*sqrt(sigma1*sigma2))}
+
+    where :math:`mu1` and :math:`sigma1` refer to the mean and covariance of the train data and
+    :math:`mu2` and :math:`sigma2` refer to the mean and covariance of the test data.
+
+    More details can be found in `Heusel et al. 2002`__.
+
+    __ https://arxiv.org/pdf/1706.08500.pdf
+
+    In addition, a faster and online computation approach can be found in `Chen et al. 2014`__
+
+    __ https://arxiv.org/pdf/2009.14075.pdf
+
+    Remark:
+
+        This implementation is inspired by pytorch_fid package.
+
+    Args:
+        num_features: specifies number of features the evaluation samples should have
+        feature_extractor: A Callable Object for extracting features from input data.
+        output_transform: a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+            By default, metrics require the output as ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+        device: specifies which device updates are accumulated on. Setting the
+            metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
+            non-blocking. By default, CPU.
+
+    Example:
+
+    .. code-block:: python
+        from ignite.metric.GAN import FID
+        import torch
+
+        train, test = torch.rand(10,2048), torch.rand(10,2048)
```

Comment:
Is it relevant as shape for the standard FID metric ?
Namings as train, test seem to be unconventional, we can put either put `y_pred`/`y` or `fake`/`real`

Reply:
FID is not just for image generation but also augmentation, style and feature transfer. In such cases real or fake is not appropriate. I will use `y_pred`/`y` instead.